pkg/cmd/server/origin/asset_apiserver.go

```diff
@@ -0,0 +1,404 @@
+	overrideConfig, err := checkForOverrideConfig(masterConfigOptions.AdmissionConfig)
+	if err != nil {
+		return nil, err
+	}
+	if overrideConfig != nil {
+		return overrideConfig, nil
+	}
+	if masterConfigOptions.KubernetesMasterConfig == nil { // external kube gets you a nil pointer here
+		return nil, nil
+	}
+	overrideConfig, err = checkForOverrideConfig(masterConfigOptions.KubernetesMasterConfig.AdmissionConfig)
+	if err != nil {
+		return nil, err
+	}
+	return overrideConfig, nil
+}
+
+// checkForOverrideConfig looks for ClusterResourceOverrideConfig plugin cfg in the admission PluginConfig
+func checkForOverrideConfig(ac configapi.AdmissionConfig) (*clusterresourceoverrideapi.ClusterResourceOverrideConfig, error) {
+	overridePluginConfigFile, err := pluginconfig.GetPluginConfigFile(ac.PluginConfig, clusterresourceoverrideapi.PluginName, "")
+	if err != nil {
+		return nil, err
+	}
+	if overridePluginConfigFile == "" {
+		return nil, nil
+	}
+	configFile, err := os.Open(overridePluginConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	overrideConfig, err := clusterresourceoverride.ReadConfig(configFile)
+	if err != nil {
+		return nil, err
+	}
+	return overrideConfig, nil
+}
+
+// Complete fills in any fields not set that are required to have valid data. It's mutating the receiver.
+func (c *AssetServerConfig) Complete() completedAssetServerConfig {
+	c.GenericConfig.Complete()
+
+	return completedAssetServerConfig{c}
+}
+
+// SkipComplete provides a way to construct a server instance without config completion.
+func (c *AssetServerConfig) SkipComplete() completedAssetServerConfig {
+	return completedAssetServerConfig{c}
+}
+
+func (c completedAssetServerConfig) New(delegationTarget genericapiserver.DelegationTarget) (*AssetServer, error) {
+	genericServer, err := c.AssetServerConfig.GenericConfig.SkipComplete().New("openshift-non-api-routes", delegationTarget) // completion is done in Complete, no need for a second time
+	if err != nil {
+		return nil, err
+	}
+
+	s := &AssetServer{
+		GenericAPIServer: genericServer,
+		IsStandalone:     c.IsStandalone,
+		PublicURL:        c.PublicURL,
+	}
+
+	if err := c.addAssets(s.GenericAPIServer.Handler.NonGoRestfulMux); err != nil {
+		return nil, err
+	}
+	if err := c.addExtensionScripts(s.GenericAPIServer.Handler.NonGoRestfulMux); err != nil {
+		return nil, err
+	}
+	if err := c.addExtensionStyleSheets(s.GenericAPIServer.Handler.NonGoRestfulMux); err != nil {
+		return nil, err
+	}
+	c.addExtensionFiles(s.GenericAPIServer.Handler.NonGoRestfulMux)
+	if err := c.addWebConsoleConfig(s.GenericAPIServer.Handler.NonGoRestfulMux); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+// buildHandlerChainForAssets is the handling chain used to protect the asset server.  With no secret information to protect
+// the chain is very short.
+func buildHandlerChainForAssets(consoleRedirectPath string) func(startingHandler http.Handler, c *genericapiserver.Config) http.Handler {
```

Comment:
> I'm a little worried that with this approach we'll have so many places to update when we'll be changing/updating/etc the handler chains.

When you run it separately, you have to have your own 'protection" in the chain.  There isn't much of an alternative.